community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/setup.py

```diff
@@ -231,17 +215,51 @@ def setup_jwt_key():
     util.chown_slurm(jwt_key, mode=0o400)
 
 
-def setup_munge_key():
-    munge_key = Path(dirs.munge / "munge.key")
+def _generate_key(p: Path) -> None:
+    run(f"dd if=/dev/random of={p} bs=1024 count=1")
 
-    if munge_key.exists():
-        log.info("Munge key already exists. Skipping key generation.")
-    else:
-        run(f"dd if=/dev/random of={munge_key} bs=1024 count=1")
 
-    shutil.chown(munge_key, user="munge", group="munge")
-    os.chmod(munge_key, stat.S_IRUSR)
-    run("systemctl restart munge", timeout=30)
+def setup_key(lkp: util.Lookup) -> None:
+    file_name = "munge.key"
+    dir = dirs.munge
+
+    if lkp.cfg.enable_slurm_auth:
+      file_name = "slurm.key"
+      dir = slurmdirs.etc
+
+    dst = Path(dir / file_name)
+
+    if lkp.cfg.controller_state_disk.device_name:
+        # Copy key from persistent state disk
+        persist = slurmdirs.state / file_name
+        if not persist.exists():
+            _generate_key(persist)
+
+        shutil.copyfile(persist, dst)
+        if lkp.cfg.enable_slurm_auth:
+            util.chown_slurm(dst, mode=0o400)
+            util.chown_slurm(persist, mode=0o400)
+        else:
+            shutil.chown(dst, user="munge", group="munge")
+            os.chmod(dst, stat.S_IRUSR)
+    else:
+        if dst.exists():
+            log.info("key already exists. Skipping key generation.")
+        else:
+            _generate_key(dst)
+            if lkp.cfg.enable_slurm_auth:
+              util.chown_slurm(dst, mode=0o400)
+            else:
+              shutil.chown(dst, user="munge", group="munge")
+              os.chmod(dst, stat.S_IRUSR)
+
+    if lkp.cfg.enable_slurm_auth:
+        # Put key into shared volume for distribution
+        distributed = util.slurmdirs.key_distribution / file_name
+        shutil.copyfile(dst, distributed)
+        util.chown_slurm(distributed, mode=0o400)
```

Comment:
nit. please add a comment that munge is also getting distributed, but straight from `etc/munge`